src/main/telemetry/ibus.c

```diff
@@ -82,23 +71,108 @@ static portSharing_e ibusPortSharing;
 
 static uint8_t ibusReceiveBuffer[IBUS_RX_BUF_LEN] = { 0x0 };
 
-
-
 static void pushOntoTail(uint8_t buffer[IBUS_MIN_LEN], size_t bufferLength, uint8_t value)
 {
     memmove(buffer, buffer + 1, bufferLength - 1);
     ibusReceiveBuffer[bufferLength - 1] = value;
 }
 
-
 void initIbusTelemetry(void)
 {
     ibusSerialPortConfig = findSerialPortConfig(FUNCTION_TELEMETRY_IBUS);
+    uint8_t type = telemetryConfig()->ibusTelemetryType;
+    uint8_t speed = type & IBUS_TYPE_SPEED_FLAG;
```

Comment:
Ok, the defines make the code a lot more understandable, thanks. Unfortunately, this also shows that `telemetryConfig()->ibusTelemetryType` is done in a very unfavourable way. Instead of packing type and speed into the same parameter, can you separate this into `telemetryConfig()->ibus_telemetry_type` (as a lookup if possible) and `telemetryConfig()->ibus_telemetry_use_m_s` (as an on / off lookup) . (As opposed to the normal naming rules for variables in cleanflight, parameter group values are named to match the name that is used for them in CLI.)